File: ios/Sources/LiveUpdateProviderMock/MockProviderConfig.swift
import LiveUpdateProvider

public struct MockProviderConfig {
    /// When true, `sync()` fails so callers can exercise error handling.
    let simulateFailure: Bool

    /// Metadata returned with a successful sync result.
    let metadata: [String: Any]

    public init(_ config: [String: Any]) throws {
        if let rawFailure = config["simulateFailure"] {
            guard let simulateFailure = rawFailure as? Bool else {
                throw ProviderError.invalidConfiguration(message: "simulateFailure must be a boolean")
            }
            self.simulateFailure = simulateFailure
        } else {
            self.simulateFailure = false
        }

        if let rawMetadata = config["metadata"] {
            guard let metadata = rawMetadata as? [String: Any] else {
                throw ProviderError.invalidConfiguration(message: "metadata must be an object")
            }
            self.metadata = metadata
        } else {
            self.metadata = [:]
        }
    }
}

File: android/live-update-provider-mock/src/main/kotlin/io/ionic/liveupdateprovidermock/MockProviderConfig.kt
package io.ionic.liveupdateprovidermock

import io.ionic.liveupdateprovider.ProviderError

data class MockProviderConfig(
    /** When true, [MockProviderManager.sync] fails so callers can exercise error handling. */
    val simulateFailure: Boolean,
    /** Metadata returned with a successful sync result. */
    val metadata: Map<String, Any>
) {
    companion object {
        fun from(config: Map<String, Any>): MockProviderConfig {
            val simulateFailure = when (val raw = config["simulateFailure"]) {
                null -> false
                is Boolean -> raw
                else -> throw ProviderError.InvalidConfiguration("simulateFailure must be a boolean")
            }

            @Suppress("UNCHECKED_CAST")
            val metadata = when (val raw = config["metadata"]) {
                null -> emptyMap()
                is Map<*, *> -> raw as Map<String, Any>
                else -> throw ProviderError.InvalidConfiguration("metadata must be an object")
            }

            return MockProviderConfig(
                simulateFailure = simulateFailure,
                metadata = metadata
            )
        }
    }
}
